app/src/main/java/in/testpress/testpress/authenticator/NewUserRegistrationActivity.java

```diff
@@ -0,0 +1,215 @@
+package in.testpress.testpress.authenticator;
+
+import android.app.Activity;
+import android.app.Dialog;
+import android.app.ProgressDialog;
+import android.content.Intent;
+import android.os.Bundle;
+import android.text.Editable;
+import android.text.TextWatcher;
+import android.view.KeyEvent;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.TextView;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.inject.Inject;
+
+import butterknife.ButterKnife;
+import butterknife.InjectView;
+import butterknife.OnClick;
+import in.testpress.testpress.Injector;
+import in.testpress.testpress.R;
+import in.testpress.testpress.R.id;
+import in.testpress.testpress.core.TestpressService;
+import in.testpress.testpress.models.RegistrationSuccessResponse;
+import in.testpress.testpress.models.RegistrationErrorDetails;
+import in.testpress.testpress.ui.TextWatcherAdapter;
+import in.testpress.testpress.util.SafeAsyncTask;
+import retrofit.RetrofitError;
+
+import static android.view.inputmethod.EditorInfo.IME_ACTION_DONE;
+
+public class NewUserRegistrationActivity extends Activity {
+    @Inject TestpressService testpressService;
+    @InjectView(id.et_username) EditText usernameText;
+    @InjectView(id.et_password) EditText passwordText;
+    @InjectView(id.et_password_confirm) EditText confirmPasswordText;
+    @InjectView(id.et_email) EditText emailText;
+    @InjectView(id.et_phone) EditText phoneText;
+    @InjectView(id.tv_fill_all_details) TextView fillAllDetailsText;
+    @InjectView(id.b_register) Button registerButton;
+    private final TextWatcher watcher = validationTextWatcher();
+    private RegistrationSuccessResponse registrationSuccessResponse;
+
+    @Override
+    public void onCreate(Bundle bundle) {
+        super.onCreate(bundle);
+        Injector.inject(this);
+        setContentView(R.layout.new_user_register_activity);
+        ButterKnife.inject(this);
+        confirmPasswordText.setOnEditorActionListener(new TextView.OnEditorActionListener() {
+            public boolean onEditorAction(final TextView v, final int actionId,
+                                          final KeyEvent event) {
+                if (actionId == IME_ACTION_DONE && registerButton.isEnabled()) {
+                    if (validate())
+                        postDetails();
+                    return true;
+                }
+                return false;
+            }
+        });
+        usernameText.addTextChangedListener(watcher);
+        passwordText.addTextChangedListener(watcher);
+        emailText.addTextChangedListener(watcher);
+        phoneText.addTextChangedListener(watcher);
+        confirmPasswordText.addTextChangedListener(watcher);
+    }
+
+    void postDetails(){
+        SafeAsyncTask<Boolean> authenticationTask;
```

Comment:
This variable is not needed. Just call new SafeAsyncTask() and finally end with execute.